### PR TITLE
Py3k: simple compat stuff

### PR DESCRIPTION
**Pull request by QuLogic**
Just a couple minor things that don't require any extra libraries.

### CODE REVIEW

File: lib/iris/_merge.py [hunk 1]
```diff
@@ -1319,10 +1319,10 @@ def axis_and_name(name):
                     # scalars for each source cube. The keys of an OrderedDict
                     # are used to retain the ordering of source cubes but to
                     # remove any duplicate tuples.
-                    cells = OrderedDict(
+                    cells = list(OrderedDict(
```
Comment (rhattersley) [inline lib/iris/_merge.py]: Do we need this one?

File: lib/iris/io/format_picker.py [hunk 2]
```diff
@@ -208,6 +208,12 @@ def handler(self):
         """The handler function of this FileFormat. (Read only)"""
         return self._handler
 
+    def __lt__(self, other):
```
Comment (rhattersley) [inline lib/iris/io/format_picker.py]: Presumably we can get rid of `__cmp__` even for Python 2?

File: lib/iris/tests/runner/_runner.py [hunk 3]
```diff
@@ -80,6 +80,7 @@ def image_as_base64(fname):
     return '\n'.join(html)
 
 
+# NOTE: Do not inherit from object as distutils does not like it.
```
Comment (rhattersley) [inline lib/iris/tests/runner/_runner.py]: :scream: Not much we can do about that I guess!

File: lib/iris/util.py [hunk 4]
```diff
@@ -862,6 +862,12 @@ def __cmp__(self, other):
             result = NotImplemented
         return result
 
+    def __lt__(self, other):
```
Comment (rhattersley) [inline lib/iris/util.py]: As elsewhere, it'd be good to get rid of `__cmp__`. In both cases it'd be good to define the other rich-comparison methods as well, that way we can keep the behaviour the same.

**Comment by rhattersley** [comment]
Thanks @QuLogic - a very useful step forwards and an easy to read PR. :smile:

File: lib/iris/_merge.py (hunk 1 above)
Comment (QuLogic) [inline lib/iris/_merge.py]: It looks like the `iter` produced from `.keys()` is smart enough to handle the two `len` before the `enumerate`, so probably not.

File: lib/iris/io/format_picker.py (hunk 2 above)
Comment (QuLogic) [inline lib/iris/io/format_picker.py]: I was only aiming for sorting in Python 3, but I think that's a possibility.

**Comment by QuLogic** [comment]
Updated.

File: lib/iris/io/format_picker.py [hunk 5]
```diff
@@ -148,6 +149,7 @@ def get_spec(self, basename, buffer_obj):
         raise ValueError(msg)
 
 
+@functools.total_ordering
```
Comment (rhattersley) [inline lib/iris/io/format_picker.py]: :+1: :smile:

File: lib/iris/util.py [hunk 6]
```diff
@@ -849,18 +851,13 @@ def __eq__(self, other):
         return (isinstance(other, type(self)) and
                 self._identity() == other._identity())
 
-    def __ne__(self, other):
```
Comment (rhattersley) [inline lib/iris/util.py]: We still need this for Python 2, otherwise the behaviour of `!=` will change. (`functools.total_ordering` doesn't help here.)

**Comment by rhattersley** [comment]
Other than the missing `__ne__` method this looks good to go.

**Comment by QuLogic** [comment]
OK!

**Comment by rhattersley** [comment]
Thanks @QuLogic :+1:

**Pull request merged**